src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -965,15 +965,9 @@ void ObjectCopyRequest<I>::compute_zero_ops() {
       if (z.get_start() + z.get_len() >= end_size) {
         // zero interval at the object end
         if (z.get_start() < prev_end_size) {
-          if (z.get_start() == 0) {
-            m_write_ops[src_snap_seq]
-              .emplace_back(COPY_OP_TYPE_REMOVE, 0, 0, 0);
-            ldout(m_cct, 20) << "COPY_OP_TYPE_REMOVE" << dendl;
-          } else {
-            m_write_ops[src_snap_seq]
-              .emplace_back(COPY_OP_TYPE_TRUNC, 0, z.get_start(), 0);
-            ldout(m_cct, 20) << "COPY_OP_TYPE_TRUNC " << z.get_start() << dendl;
-          }
+          m_write_ops[src_snap_seq]
```

Comment:
Nit: perhaps just put a ```TODO``` since if you are flattening, you can send the remove.

Reply:
@dillaman 
> In this case, if all your source objects are empty, you would just need to issue a create+truncate if there is a parent overlap, correct?

That means that if we have a large parent image with only one object used, clone it, then deep copy will create many empty objects (for every nonexistent object in parent)?

Reply:
> On discard we do "create+truncate" for this object, so it hides the parent. But when deep copying the object calc_snap_set_diff returns that it does not exist

Actually -- thinking about this, if the original clone had to issue a ```create+truncate```, why would ```calc_snap_set_diff``` return ```-ENOENT```? The object should exist, but it should be size zero. When deep-copying this object, if this is a clone w/ parent overlap, you would just need to preserve the ```create```, right?

Reply:
Hmm -- sounds like a bug in ```calc_snap_set_diff```. If retrieving the ```snap_set_t``` worked, it means the object exists. If the object doesn't have any clones, it means the object exists throughout the full history of the object.